### ISSUE TITLE
Profile picture upload is not working

### ISSUE THREAD

**Issue by ani4aniket** [RocketChat/Rocket.Chat.ReactNative#1409]
I have attached a screen recording which I've faced during running the app on iOS simulator.
![Screen Recording 2019-11-21 at 5 23 41 PM](https://user-images.githubusercontent.com/31184591/69336576-cf382100-0c84-11ea-9a79-573c467f9b57.gif)

**Comment by diegolmello** [comment]
@ani4aniket There're two issues apparently.
First one is not showing the image you've selected at the top of the view.
The other is the avatar cache not refreshing, because your avatar seems to be changed on the web.
<img src="https://user-images.githubusercontent.com/804994/69338943-0b4a9200-0c43-11ea-9c08-1ecfbfcb5614.png" width="350" />

**Comment by ani4aniket** [comment]
Yeah, You are right as in my phone app. The image is not updated. 
![Screenshot_2019-11-21-18-54-16-04_6c2112e95a10afaab76cfa27a6d42c7b](https://user-images.githubusercontent.com/31184591/69341961-8ab28280-0c90-11ea-8ee6-e7d0712becdc.png)

**Comment by nccurry** [comment]
Seeing this issue on android as well

**Comment by mahal** [comment]
As by my tests, this is related to the caching issue #441 

But by my tests I encountered following when uploading avatars from both Android and iOS:

Steps to reproduce:
- Start RC-RN client 
- Login to open.rocket.chat
- Tap Menu > Profile
- Choose image from picture roll 
- Save Changes

Actual result:
- App shows "There was an error while changig avatar!"

Environment:
RC-RN client: 4.6.4.295 iOS, 4.6.4.2118 (Android)

The same happens on our own rocket.chat server: 2.4.11 
The log file shows
```
I20200422-09:08:32.874(0) Sanitized and reported to the client as: { Error: Match failed [400]     at errorClass.<anonymous> (packages/check/match.js:91:27)     at new errorClass (packages/meteor.js:655:17)     at check (packages/check/match.js:36:17)     at MethodInvocation.setAvatarFromService (server/methods/setAvatarFromService.js:15:3)     at MethodInvocation.methodsMap.(anonymous function) (app/lib/server/lib/debug.js:67:34)     at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1771:12)     at DDP._CurrentMethodInvocation.withValue (packages/ddp-server/livedata_server.js:719:19)     at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)     at DDPServer._CurrentWriteFence.withValue (packages/ddp-server/livedata_server.js:717:46)     at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)     at Promise (packages/ddp-server/livedata_server.js:715:46)     at new Promise (<anonymous>)     at Session.method (packages/ddp-server/livedata_server.js:689:23)     at packages/ddp-server/livedata_server.js:559:43   isClientSafe: true,   error: 400,   reason: 'Match failed',   details: undefined,   message: 'Match failed [400]',   errorType: 'Meteor.Error' } 
I20200422-09:08:32.874(0) undefined undefined 
```

**Comment by nofikoff** [comment]
I compared the logs of the upload avatar from the web browser and from the mobile application.
When generating a request from a mobile application, the last parameter is "{}" and must be "" or empty!

Adnroid APP with error upload:
server.js:204 Meteor ➔ method setAvatarFromService -> userId: aW7eNTaqHETbp5YnK, arguments: [“data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAgGBgcGBQ< … >ge4zz9azoXIJxipERZZTvGaBlpSiy4hkZyOeP8DVj7ZcDja3/fIrN/1ch28VVe5l3H5jUgf//Z”,
“”,“upload”,{}]

Chrome Browser success upload:
server.js:204 Meteor ➔ method setAvatarFromService -> userId: aW7eNTaqHETbp5YnK, arguments: [“data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD/2wBDAAgGBgcGBQ< … >ge4zz9azoXIJxipERZZTvGaBlpSiy4hkZyOeP8DVj7ZcDja3/fIrN/1ch28VVe5l3H5jUgf//Z”,
“image/jpeg”,“upload”]

**Comment by djorkaeffalexandre** [comment]
Hey @nofikoff, I think your error was fixed by #2050. It'll be launched as a public beta soon, try to enroll on our beta apps.
Thanks in advance.
